numpy/core/src/multiarray/arraytypes.c.src

```diff
@@ -2748,6 +2748,15 @@ OBJECT_compare(PyObject **ip1, PyObject **ip2, PyArrayObject *NPY_UNUSED(ap))
      * the alignment of pointers, so it doesn't need to be handled
      * here.
      */
+
+    int ret;
+    /*
+     * work around gh-3879, we cannot abort an in-progress quicksort
+     * so at least do not raise again
+     */
+    if (PyErr_Occurred()) {
+        return 1;
```

Comment:
Rationale for choosing this over `0`? I'd worry a sort could hang if 1 is returned, since the comparisons become inconsistent (`cmp(a, b) == cmp(b, a) == 1`)